### PR TITLE
Update XML docs for OnStarting

### LINKED ISSUE #19530: httpContext.Response.OnStarting lambdas executing in reverse order
I have a setup where the same cookie might get deleted, set or changed in the same request based on various business logic. As per Microsoft suggestion, I have these cookie updates wrapped up inside OnStarting. In doing so, I have noticed that the lamdas run in the REVERSE order of their registration.

Eg: 
```

            this.HttpContext.Response.OnStarting(
                () =>
                {
                    Console.WriteLine(1);
                    return Task.CompletedTask;
                });

            this.HttpContext.Response.OnStarting(
               () =>
               {
                   Console.WriteLine(2);
                   return Task.CompletedTask;
               });

            this.HttpContext.Response.OnStarting(
               () =>
               {
                   Console.WriteLine(3);
                   return Task.CompletedTask;
               });

            this.HttpContext.Response.OnStarting(
               () =>
               {
                   Console.WriteLine(4);
                   return Task.CompletedTask;
               });

            this.HttpContext.Response.OnStarting(
               () =>
               {
                   Console.WriteLine(5);
                   return Task.CompletedTask;
               });

            this.HttpContext.Response.OnStarting(
               () =>
               {
                   Console.WriteLine(6);
                   return Task.CompletedTask;
               });

            this.HttpContext.Response.OnStarting(
               () =>
               {
                   Console.WriteLine(7);
                   return Task.CompletedTask;
               });

```

Outputs:
7
6
5
4
3
2
1

ASP.NET Core version: 3.1

### PR DESCRIPTION
**Pull request by analogrelay**
Fixes https://github.com/dotnet/aspnetcore/issues/19530

I have both a brief `<summary>` update (because that shows in IntelliSense) and detailed justification in `<remarks>`.

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

File: src/Http/Http.Abstractions/src/HttpResponse.cs [hunk 1]
```diff
@@ -72,14 +72,26 @@ public abstract class HttpResponse
 
         /// <summary>
         /// Adds a delegate to be invoked just before response headers will be sent to the client.
+        /// Callbacks registered here will be run in reverse order.
         /// </summary>
+        /// <remarks>
+        /// Callbacks registered here will be run in reverse order (the last one registered will be invoked first).
+        /// This is done to replicate the way middleware works (with the inner-most middleware looking at the response
+        /// first).
```
Comment (Rick-Anderson) [inline src/Http/Http.Abstractions/src/HttpResponse.cs]:
```suggestion
        /// Callbacks registered here run in reverse order.
        /// </summary>
        /// <remarks>
        /// Callbacks registered here run in reverse order. The last one registered is invoked first.
        /// The reverse order is done to replicate the way middleware works, with the inner-most middleware looking at the
        /// response first).
```

File: src/Http/Http.Abstractions/src/HttpResponse.cs [hunk 2]
```diff
@@ -72,14 +72,26 @@ public abstract class HttpResponse
...
+        /// </remarks>
```
Comment (Rick-Anderson) [inline src/Http/Http.Abstractions/src/HttpResponse.cs]:
```suggestion
        /// Callbacks registered here run in reverse order.
        /// </summary>
        /// <remarks>
        /// Callbacks registered here run in reverse order. The last one registered is invoked first.
        /// The reverse order is done to replicate the way middleware works, with the inner-most middleware looking at the
        /// response first).
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]
Per MS style guide:

* Avoid future tense. Future tense is not banned, but when not needed avoid it.
* Avoid parenthetical clauses. Again, not banned.

**Comment by anurse** [comment]
Thanks @Rick-Anderson !

File: src/Http/Http.Abstractions/src/HttpResponse.cs [hunk 3]
```diff
@@ -72,14 +72,25 @@ public abstract class HttpResponse
...
         public abstract void OnStarting(Func<object, Task> callback, object state);
 
         /// <summary>
         /// Adds a delegate to be invoked just before response headers will be sent to the client.
+        /// Callbacks registered here run in reverse order.
         /// </summary>
+        /// <remarks>
+        /// Callbacks registered here run in reverse order. The last one registered is invoked first.
+        /// The reverse order is done to replicate the way middleware works, with the inner-most middleware looking at the
+        /// response first.
```
Comment (anurse) [inline src/Http/Http.Abstractions/src/HttpResponse.cs]:
```suggestion
        /// response first.
        /// </remarks>
```

**Comment by anurse** [comment]
Rebased

**Pull request merged**